src/main/java/com/vw/ipppdiffer/service/DifferServiceImpl.java

```diff
@@ -41,43 +40,31 @@
     private final DifferComponent differComponent;
 
     @Override
-    public DifferResponse getDiffer(MultipartFile firstFile, MultipartFile secondFile) {
+    public DifferResponse getDiffer(MultipartFile firstFile, MultipartFile secondFile) throws JAXBException, IOException, IllegalAccessException {
         Element firstTree = buildTreeFromFile(firstFile, ColourType.NONE);
         Element secondTree = buildTreeFromFile(secondFile, ColourType.NONE);
         differComponent.compareTrees(singletonList(firstTree), singletonList(secondTree));
         return new DifferResponse(firstTree, secondTree);
     }
 
     @Override
-    public Element getTreeStructure(MultipartFile file) {
+    public Element getTreeStructure(MultipartFile file) throws IOException, IllegalAccessException, JAXBException {
         return buildTreeFromFile(file, ColourType.BLACK);
     }
 
     @SuppressWarnings("unchecked")
-    private IB1 parseXMLFile(InputStream xmlFile) {
-        try {
-            JAXBContext jaxbContext = JAXBContext.newInstance(ObjectFactory.class);
-            return ((JAXBElement<IB1>) jaxbContext.createUnmarshaller().unmarshal(xmlFile)).getValue();
-        } catch (JAXBException e) {
-            throw new UnknownException("Could not deserialize XML");
-        }
+    private IB1 parseXMLFile(InputStream xmlFile) throws JAXBException {
+        JAXBContext jaxbContext = JAXBContext.newInstance(ObjectFactory.class);
+        return ((JAXBElement<IB1>) jaxbContext.createUnmarshaller().unmarshal(xmlFile)).getValue();
+
     }
 
-    private Element buildTreeFromFile(MultipartFile file, ColourType colour) {
+    private Element buildTreeFromFile(MultipartFile file, ColourType colour) throws IOException, IllegalAccessException, JAXBException {
         InputStream fileInputStream;
-        try {
-            fileInputStream = file.getInputStream();
-        } catch (IOException e) {
-            throw new UnknownException("Could not read xml file");
```

Comment:
It was correct to throw your own exception from the service. Change to throw your newly created exception